cea/optimization/slave/cooling_resource_activation.py

```diff
@@ -144,268 +322,265 @@ def cooling_resource_activator(Q_thermal_req,
+            Qc_from_storage_W = 0.0
+            Qc_VCC_WS_gen_W = Qc_VCC_WS_gen_directload_W + Qc_VCC_WS_gen_storage_W
+
+        # Activation Case 3: The water temperature allows for free cooling, therefore the VCC is bypassed.
+        # The unmet cooling demand is larger than the water body's cooling potential, hence the storage is discharged.
+        elif T_district_cooling_supply_K - T_source_average_water_body_K >= DT_COOL \
+                and Q_cooling_unmet_W >= Qc_water_body_remaining_W:
+            FreeCooling_WS_activated = True
+            Qc_FreeCooling_WS_directload_W = Qc_water_body_remaining_W
+            Qc_water_body_to_storage_W = 0.0
+            Qc_from_storage_W, Qc_DailyStorage_content_W = \
+                daily_storage_class.discharge_storage(Q_cooling_unmet_W - Qc_water_body_remaining_W)
+            Qc_FreeCooling_and_DirectStorage_WS_W = Qc_FreeCooling_WS_directload_W + Qc_water_body_to_storage_W
+
+        # Activation Case 4: The water temperature allows for free cooling, therefore the VCC is bypassed.
+        # The water body's cooling potential is larger than the unmet cooling demand, therefore the storage is charged.
+        elif T_district_cooling_supply_K - T_source_average_water_body_K >= DT_COOL \
+                and Q_cooling_unmet_W < Qc_water_body_remaining_W:
+            FreeCooling_WS_activated = True
+            Qc_FreeCooling_WS_directload_W = Q_cooling_unmet_W
+            Qc_water_body_to_storage_W, Qc_DailyStorage_content_W = \
+                daily_storage_class.charge_storage(Qc_water_body_remaining_W - Q_cooling_unmet_W)
             Qc_from_storage_W = 0.0
-            Q_PeakVCC_WS_gen_W = Q_PeakVCC_WS_gen_directload_W + Qc_PeakVCC_WS_gen_storage_W
-
-        # If water body temperature is higher than the temperature required in the district cooling network's heat
-        # exchanger, use the vapour compression chiller to bridge the temperature difference...
-        if (T_district_cooling_supply_K - DT_COOL) < T_source_average_water_body_K < VCC_T_COOL_IN:
-            WS_PeakVCC_capacity = master_to_slave_variables.WS_PeakVCC_size_W
-            Q_PeakVCC_WS_gen_W, \
-            E_PeakVCC_WS_req_W = calc_vcc_operation(Q_PeakVCC_WS_gen_W,
-                                                    T_district_cooling_return_K,
-                                                    T_district_cooling_supply_K,
-                                                    T_source_average_water_body_K,
-                                                    WS_PeakVCC_capacity,
-                                                    VC_chiller)
-            E_pump_WS_req_W = calc_water_body_uptake_pumping(Q_PeakVCC_WS_gen_W,
+            Qc_FreeCooling_and_DirectStorage_WS_W = Qc_FreeCooling_WS_directload_W + Qc_water_body_to_storage_W
+
+        # Determine the electricity needed for the hydraulic pumps and the VCC if the latter is activated...
```

Comment:
# Determine the electricity needed for the hydraulic pumps and the VCC if the latter is activated.

Reply:
Do you mean I should remove the three dots at the end of the line?